Library/Homebrew/cmd/cat.rb

```diff
@@ -25,6 +26,18 @@ def cat
     raise "`brew cat` doesn't support multiple arguments" if args.remaining.size > 1
 
     cd HOMEBREW_REPOSITORY
-    safe_system "cat", formulae.first.path, *Homebrew.args.passthrough
+    if ENV["HOMEBREW_BAT"].nil?
+      pager = "cat"
+    else
+      name = "bat"
+      pager = "#{Homebrew::DEFAULT_PREFIX}/bin/#{name}"
+      unless Formula[name].installed?
+        fi = FormulaInstaller.new(Formula[name])
+        fi.prelude
+        fi.install
+        fi.finish
```

Comment:
What happens if this installation fails? I think it'll just get reattempted on the next execution of `brew cat` but perhaps it should fall back to `cat` in that case.

Reply:
Should I wrap this up in `begin` and `rescue`?

Reply:
I like how you re-did this! 🎉